prometheus/histogram.go

```diff
@@ -1835,3 +1835,129 @@ func (n *nativeExemplars) addExemplar(e *dto.Exemplar) {
 		n.exemplars = append(n.exemplars[:nIdx], append([]*dto.Exemplar{e}, append(n.exemplars[nIdx:rIdx], n.exemplars[rIdx+1:]...)...)...)
 	}
 }
+
+type constNativeHistogram struct {
+	desc                            *Desc
+	count                           uint64
+	sum                             float64
+	labelPairs                      []*dto.LabelPair
+	nativeHistogramSchema           int32
+	nativeHistogramZeroThreshold    float64
+	nativeHistogramMaxZeroThreshold float64
+	nativeHistogramMaxBuckets       uint32
+	nativeHistogramMinResetDuration time.Duration
+	timeStamp                       time.Time
+	nativeExemplars                 []*dto.Exemplar
+
+	positiveBuckets map[int]int64
+	negativeBuckets map[int]int64
+	zeroBucket      uint64
+}
+
+func NewconstNativeHistogram(desc *Desc, count uint64, sum float64, postiveBuckets, negativeBuckets map[int]int64, zeroBucket uint64,
```

Comment:
```suggestion
func NewConstNativeHistogram(desc *Desc, count uint64, sum float64, postiveBuckets, negativeBuckets map[int]int64, zeroBucket uint64,
```

Reply:
Could you also add.MustNewConstNativeHistogram that re uses the one you just added and panics if there's an error?

This would be consistent with all the other NewCost... functions that we have :)